enhancements/sig-architecture/11-cluster-deletion/README.md

```diff
@@ -0,0 +1,122 @@
+
+   
+### User Stories
+
+#### Story 1
+
+As a cluster admin, the cluster namespace should not be deleted after I delete the `managedCluster`, because I deployed other resources in this namespace.
+
+  Prerequisites:
+   1. Add the label `cluster.open-cluster-management.io/delete-by-other` to the cluster namespace.
+   
+  Deletion process:
+   1. The cluster admin deletes the `managedCluster` on the hub.
+   2. `deletion-controller` deletes all `managedClusterAddons`.
+   3. `deletion-controller` deletes all `manifestWorks`.
+   4. `deletion-controller` deletes `roleBindings` of registration-agent and work-agent.
+   5. `deletion-controller` force deletes the `managedClusterAddons` and `manifestWorks` after the `forceDeletionGracePeriodSeconds`.
+   6. The cluster admin deletes the `klusterlet` on the spoke cluster.
+
+#### Story 2
+
+As a cluster admin, all cluster resources should be cleaned up after I delete the `managedCluster`,including the cluster namespace on the hub cluster and the resources deployed by `manifestWork` on the spoke cluster. 
+
+   Deletion process:
+   1. The cluster admin deletes the `managedCluster` on the hub.
+   2. `deletion-controller` deletes all `managedClusterAddons`.
+   3. `deletion-controller` deletes all `manifestWorks`.
+   4. `deletion-controller` deletes `roleBindings` of registration-agent and work-agent.
+   5. `deletion-controller` force deletes the `managedClusterAddons` and `manifestWorks` after the `forceDeletionGracePeriodSeconds`.
+   6. `deletion-controller` deletes the cluster namespace.
+   7. The cluster admin deletes the `klusterlet` on the spoke cluster.
+#### Story 3 
+
+As a cluster admin, I want some special `manifestWorks` to be the last one to be deleted by myself when I delete the `managedCluster`.
+
+ Prerequisites: 
+   1. Add the label `cluster.open-cluster-management.io/delete-by-other` to the special `manifestWork`.
+   
+   Deletion process:
+   1. The cluster admin deletes the `managedCluster` on the hub.
+   2. `deletion-controller` deletes all `managedClusterAddons`.
+   3. `deletion-controller` deletes all `manifestWorks` without the label `cluster.open-cluster-management.io/delete-by-other`.
+   4. Other controller deletes the special `manifestWorks` after the other `manifestWorks` are deleted.
+   5. `deletion-controller` deletes `roleBindings` of registration-agent and work-agent.
+   6. `deletion-controller` force deletes the `managedClusterAddons` and `manifestWorks` after the `forceDeletionGracePeriodSeconds`.
+   7. `deletion-controller` deletes the cluster namespace.
```

Comment:
please add upgrade strategy for this feature. Also we should put this in a feature gates.

Reply:
done